pkg/processor/runtime/python/py/_nuclio_async_wrapper.py

```diff
@@ -68,8 +68,8 @@ def __init__(self,
         self._host = split_address[0]
         self._port = int(split_address[1])
 
-        # Max file descriptors allowed by the OS
-        self._max_connections = max_connections if max_connections else os.sysconf("SC_OPEN_MAX")
+        # 80 percent of Max file descriptors allowed by the OS
+        self._max_connections = max_connections if max_connections else int(os.sysconf("SC_OPEN_MAX") * 0.8)
```

Comment:
If someone set `max_connection` to be a ridiculously large number, we don't want to allow it.
better do something like this:
```suggestion
        # 80 percent of Max file descriptors allowed by the OS
        system_max_connections = int(os.sysconf("SC_OPEN_MAX") * 0.8)
        self._max_connections = min(max_connections, system_max_connections) if max_connections else system_max_connections
```